src/net.c

```diff
@@ -351,6 +351,18 @@ int git_net_url_fmt(git_buf *buf, git_net_url *url)
 	return git_buf_oom(buf);
 }
 
+int git_net_url_fmt_path(git_buf *buf, git_net_url *url)
+{
+	git_buf_puts(buf, url->path ? url->path : "/");
+
+	if (url->query) {
+		git_buf_putc(buf, '?');
+		git_buf_puts(buf, url->query);
+	}
+
+	return git_buf_oom(buf);
```

Comment:
This function will return positive if the buffer ran OOM.

Reply:
Oops - thanks!  Fixed.